cabot/cabotapp/tests/test_acks.py

```diff
@@ -0,0 +1,198 @@
+from django.core.urlresolvers import reverse
+from mock import patch
+from django.utils import timezone
+
+from cabot.cabotapp import tasks
+from cabot.cabotapp.models import Acknowledgement
+from .utils import LocalTestCase, fake_http_404_response, fake_http_200_response
+
+
+class TestAcks(LocalTestCase):
+    def setUp(self):
+        super(TestAcks, self).setUp()
+        self.service.update_status()
+
+        self.client.login(username=self.username, password=self.password)
+
+    def fail_http_check(self):
+        """runs self.http_check such that it will fail, then returns the StatusCheckResult"""
+        with patch('cabot.cabotapp.models.requests.request', fake_http_404_response):
+            self.http_check.run()
+
+        return self.http_check.last_result()
+
+    def pass_http_check(self):
+        """runs self.http_check such that it will pass, then returns the StatusCheckResult"""
+        with patch('cabot.cabotapp.models.requests.request', fake_http_200_response):
+            self.http_check.run()
+        return self.http_check.last_result()
+
+    def test_create_ack_for_result(self):
+        """tests creating an ack using the create-ack endpoint with the result_id param (submits default form)"""
+        result = self.fail_http_check()
+        self.assertFalse(result.succeeded)
+        self.assertFalse(result.acked)
+
+        # test create ack page pre-filled with result id
+        url = '{}?result_id={}'.format(reverse('create-ack'), result.id)
+        data = self.client.get(url).context['form'].initial  # the data to post is what's pre-filled
+        resp = self.client.post(url, data=data)  # post it
+        self.assertEquals(resp.status_code, 302)
+
+        # make sure it created the ack we expect
+        ack = Acknowledgement.objects.get(status_check=self.http_check)
+        self.assertEquals(ack.created_by_id, self.user.id)
+        self.assertAlmostEqual((ack.expire_at - ack.created_at).seconds, 4 * 60 * 60)  # expiry defaults to 4 hours
```

Comment:
If I remember correctly, I intentionally didn't freeze time here because it wasn't freezing Django's "default to now()" behavior. But yeah, I'll tighten this up.